include/mfmg/dealii_operator.hpp

```diff
@@ -114,6 +119,25 @@ class DealIITrilinosMatrixOperator : public MatrixOperator<VectorType>
   std::shared_ptr<matrix_type> _matrix;
 };
 
+template <typename VectorType>
+class DealIITrilinosMatrixfreeOperator
```

Comment:
A better name for this would be `DealIIMatrixFreeOperator` because ideally we won't have any matrices except at the coarsest level.